src/participant/sampleDocuments.ts

```diff
@@ -0,0 +1,31 @@
+import type { Document } from 'bson';
+import { toJSString } from 'mongodb-query-parser';
+
+const MAX_ARRAY_LENGTH_OF_SAMPLE_DOCUMENT_VALUE = 3;
+
+const MAX_STRING_LENGTH_OF_SAMPLE_DOCUMENT_VALUE = 20;
+
+export function formatSampleDocuments(obj: Document) {
```

Comment:
Should the argument type be `Document[]` instead? Also, would it make sense for the return type to be `string[]` which is the truncated projection of the supplied document collection? Not sure if it's out of scope, but I was thinking we could rework our approach to take 0-4 documents instead of all or nothing. Don't know if it'll improve the accuracy of the responses, but figured it's not a lot of effort so we might as well do it.

Reply:
If we want to match compass behavior this would be all or nothing: https://github.com/10gen/mms/blob/master/server/src/main/com/xgen/cloud/naturallanguagequery/_public/svc/BuildQueryPromptUtils.java#L144

@Anemy maybe you can give feedback if this approach goes well so far for compass? does it worth an efforts to do 1-3?

Reply:
I'd like to test if the best behavior is only to supply one document for less content in the prompt to keep the ai from losing focus. For now I like reducing it to one document when 3 is too much for the context length, that seems like a good solution and shouldn't introduce much complexity.